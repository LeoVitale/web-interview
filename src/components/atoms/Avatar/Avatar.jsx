import React from 'react'
import PropTypes from 'prop-types'
import styles from './Avatar.module.scss'

const Avatar = ({ img, name, icon, className }) => {
  const { avatar } = styles

  return (
    <div className={`${avatar} ${className}`}>
      {icon ? icon : <img src={img} alt={name} />}
    </div>
  )
}

Avatar.propTypes = {}

export default Avatar
