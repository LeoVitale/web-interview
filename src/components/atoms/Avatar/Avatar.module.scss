@import 'styles/_constants.scss';

.avatar {
  margin-right: 20px;
  border-radius: 50%;

  display: flex;
  align-items: center;
  justify-content: center;

  img {
    border-radius: 50%;
    width: 100%;
    height: 100%;
  }
}

.small {
  width: 40px;
  height: 40px;
}

.medium {
  width: 70px;
  height: 70px;
}
