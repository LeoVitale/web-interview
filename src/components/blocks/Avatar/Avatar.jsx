import React from 'react';
import styles from './Avatar.module.scss';

const Avatar = () => {
  const { avatar } = styles;
  return (
    <div className={avatar}>
      <img src="https://react.semantic-ui.com/images/avatar/small/elliot.jpg" alt=""/>
    </div>
  );
};

export default Avatar;
