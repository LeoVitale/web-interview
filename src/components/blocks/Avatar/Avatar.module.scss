.avatar {
  margin-right: 20px;
  img {
    border-radius: 50%;
  }
}
