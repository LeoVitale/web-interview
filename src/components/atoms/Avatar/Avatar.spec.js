import React from 'react'
import { mount } from 'enzyme'
import Avatar from './Avatar.jsx'

describe('<Avatar />', () => {
  const avatar = mount(<Avatar />)

  it('should render Avatar as expected', () => {
    expect(avatar).toMatchSnapshot()
  })
})
