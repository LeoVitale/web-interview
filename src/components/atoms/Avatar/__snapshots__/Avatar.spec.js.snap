// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render Avatar as expected 1`] = `
<Avatar>
  <div
    className="avatar undefined medium"
  >
    <img />
  </div>
</Avatar>
`;
